lib/elixir/lib/file.ex

```diff
@@ -931,6 +931,11 @@ defmodule File do
   option when opening the file and then all other functions from
   `IO` are available, since they work directly with Unicode data.
 
+  `modes_or_function` can either be a list of modes or a function. If it's a
+  list, it's considered to be a list of modes (that are documented below). If
+  it's a function, then it's equivalent to calling `open(path, [],
+  modes_or_function)`. See the documentation for `open/3` for more information.
```

Comment:
Why how the code works affects it? The code could be anyway we want, we could just even delegate everything to private functions?

Anyway, the PR documentation is fine by me, I just wanted to bring this point briefly. :)